tests/integration/models/subscriptions.rs

```diff
@@ -0,0 +1,100 @@
+use crate::utils::{random_ascii_string, random_currency_or_country, random_price, spawn_app};
+use lib::models::subscriptions::{Subscription, SubscriptionModel};
+use pretty_assertions::assert_eq;
+use serde_json::json;
+use time::{Duration, OffsetDateTime};
+use uuid::Uuid;
+
+fn make_fake_sub() -> Subscription {
+    Subscription {
+        id: Uuid::new_v4(),
+        flow_id: random_ascii_string(),
+        subscription_id: random_ascii_string(),
+        report_timestamp: OffsetDateTime::now_utc(),
+        subscription_created: OffsetDateTime::now_utc() - Duration::hours(35),
+        fxa_uid: random_ascii_string(),
+        quantity: 1,
+        plan_id: random_ascii_string(),
+        plan_currency: random_currency_or_country(),
+        plan_amount: random_price(),
+        country: Some(random_currency_or_country()),
+        aic_id: Some(Uuid::new_v4()),
+        aic_expires: Some(OffsetDateTime::now_utc()),
+        cj_event_value: Some(random_ascii_string()),
+        status: Some(random_ascii_string()),
+        status_history: Some(json!([])),
+    }
+}
+
+#[tokio::test]
+async fn test_subscription_model_create_from_subscription_struct() {
+    let app = spawn_app().await;
+    let model = SubscriptionModel {
+        db_pool: &app.db_connection(),
+    };
+    let sub = make_fake_sub();
+    model
+        .create_from_sub(&sub)
+        .await
+        .expect("Failed to create test object.");
+    let result = model
+        .fetch_one_by_id(&sub.id)
+        .await
+        .expect("Could not fetch from DB.");
+    assert_eq!(result, sub);
+}
+
+#[tokio::test]
+async fn test_subscription_model_fetch_ones_by_unique_ids() {
+    let app = spawn_app().await;
+    let model = SubscriptionModel {
+        db_pool: &app.db_connection(),
+    };
+    let sub = make_fake_sub();
+    model
+        .create_from_sub(&sub)
+        .await
+        .expect("Failed to create test object.");
+    let result = model
```

Comment:
Is this necessary? You made the same checks in the test just above this.

Reply:
you can't see this because the PR is a monolith but this reflects me building up the subscriptions model one method at a time tdd style. 

note that these are effectively unit tests for the models but are in the integration tests folder because they involve actually using the database.

first i wrote a test for fetch_one_by_id; then by_flow_id; then for making a sub etc. between each i was using that one method until i needed more.

each test is designed to test the one one function. that other tests then use those methods....in my tdd brain we could delete the create_from_subscription method and associated test without needing to write any new tests to facilitate that deletion.

so....is it necessary....i'm happy to have a longer discussion about testing and where we draw the line.